src/filesystem.cpp

```diff
@@ -761,21 +756,17 @@ static bfs::path apple_userdata(const std::string& newprefdir)
 		temp = "Library/Application Support/Wesnoth_"+get_version_path_suffix();
 		DBG_FS << "userdata using default path relative to HOME";
 #endif
-	} else if(temp[0] != '/') {
-		// TRANSLATORS: translate the part inside <...> only
-		deprecated_message(_("--userdata-dir=<relative path>"),
-			DEP_LEVEL::FOR_REMOVAL,
-			{1, 17, 0},
-			_("Use absolute paths. Relative paths are deprecated because they are interpreted relative to $HOME"));
 	}
 
 	// if it's an absolute path, just use that
 	// else make it relative to HOME if HOME is populated, otherwise make it relative to the current working directory
 	if(temp[0] == '/') {
 		dir = temp;
+		DBG_FS << "userdata using absolute path";
 	} else {
-		bfs::path home = home_str ? home_str : ".";
-		dir = home / temp;
+		dir = ".";
+		dir / temp;
+		DBG_FS << "unable to determine location to use for userdata, defaulting to current working directory";
```

Comment:
Another confusing message that tells the user little and is a debug message. It needs to be clearly communicated to the user what is happening.

Reply:
Updated.